chainercv/datasets/cityscapes/cityscapes_semantic_segmentation_dataset.py

```diff
@@ -0,0 +1,85 @@
+import glob
+import os
+
+import numpy as np
+
+from chainer import dataset
+from chainercv.datasets.cityscapes.cityscapes_utils import cityscapes_labels
+from chainercv.utils import read_image
+
+
+class CityscapesSemanticSegmentationDataset(dataset.DatasetMixin):
+
+    """Dataset class for a semantic segmentation task on `Cityscapes dataset`_.
+
+    .. _`Cityscapes dataset`: https://www.cityscapes-dataset.com
+
+    .. note::
+
+        Please download the data by yourself because Cityscapes dataset doesn't
+        allow to re-distribute their data.
+
+    Args:
+        img_dir (string): Path to the image dir. It should end with
+            ``leftImg8bit``.
+        label_dir (string): Path to the dir which contains labels. It should
+            end with either ``gtFine`` or ``gtCoarse``.
+        split ({'train', 'val'}): Select from dataset splits used in
+            Cityscapes dataset.
+        ignore_labels (bool): If True, the labels marked ``ignoreInEval``
+            defined in the original
+            `cityscapesScripts<https://github.com/mcordts/cityscapesScripts>_`
+            will be replaced with `-1` in the `get_example` method.
+
+    """
+
+    def __init__(self, img_dir, label_dir, split='train', ignore_labels=True):
+        img_dir = os.path.join(img_dir, split)
+        self.ignore_labels = ignore_labels
+
+        self.label_fns, self.img_fns = [], []
+        resol = os.path.basename(label_dir)
+        for dname in glob.glob('{}/*'.format(label_dir)):
+            if split in dname:
+                for label_fn in glob.glob(
+                        '{}/*/*_labelIds.png'.format(dname)):
+                    self.label_fns.append(label_fn)
+        for label_fn in self.label_fns:
+            img_fn = label_fn.replace(resol, 'leftImg8bit')
+            img_fn = img_fn.replace('_labelIds', '')
+            self.img_fns.append(img_fn)
+
+    def __len__(self):
+        return len(self.img_fns)
+
+    def get_example(self, i):
+        """Returns the i-th example.
+
+        Returns a color image and a label image. The color image is in CHW
+        format and the label image is in HW format.
+
+        Args:
+            i (int): The index of the example.
+
+        Returns:
+            tuple of a color image and a label whose shapes are (3, H, W) and
+            (H, W) respectively. H and W are height and width of the image.
+            The dtype of the color image is :obj:`numpy.float32` and
+            the dtype of the label image is :obj:`numpy.int32`.
+
+        """
+        img = read_image(self.img_fns[i])
+        label_orig = read_image(
+            self.label_fns[i], dtype=np.int32, color=False)[0]
+        H, W = label_orig.shape
+        if self.ignore_labels:
+            label_out = np.ones((H, W), dtype=np.int32) * -1
```

Comment:
Ahh. I see. My bad.
It looks OK.
Is it OK to remove line 79?
Also, np.where is not necessary.